lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -204,13 +209,21 @@ internal void Reset()
 
         /// <summary>
         /// This method is called when receiving ICompletedTask event during task running or system shutting down.
+        /// If it is master task and if the master task was running, mark _masterTaskCompleted true
         /// Removes the task from running tasks if it was running
         /// Changes the task state from RunningTask to CompletedTask if the task was running
         /// Change the task stat from TaskWaitingForClose to TaskClosedByDriver if the task was in TaskWaitingForClose state
         /// </summary>
         /// <param name="completedTask"></param>
         internal void RecordCompletedTask(ICompletedTask completedTask)
         {
+            if (completedTask.Id.Equals(_masterTaskId))
+            {
+                if (GetTaskInfo(completedTask.Id).TaskState.CurrentState.Equals(TaskState.TaskRunning))
```

Comment:
This method is called when a task was in TaskRunning state or TaskWaitingForClose state. That is the same for master task.  So if master task was closed by driver then completed and reaches to this method, we should NOT set _masterTaskCompleted. Here _masterTaskCompleted is true means the system has successfully done, doesn't means only receives completed event from master task.